retry.go

```diff
@@ -37,24 +37,27 @@ func New(sleep time.Duration) *Retry {
 	return r
 }
 
+// ContinueOnNil makes the retry continue even if the run function
+// returns nil. You will need to set explicit conditions for the
+// retry to exit.
+func (r *Retry) ContinueOnNil() *Retry {
+	// The first post condition is the one that returns false
+	// if the error is nil so we need to remove it.
+	r.postConditions = r.postConditions[1:]
```

Comment:
This is pretty janky, but I trust that you've considered alternatives.

Reply:
Did not actually, I agree its pretty janky. I just did it like that to save time. I'll fix.